lightning/src/util/events.rs

```diff
@@ -68,6 +69,53 @@ pub enum PaymentPurpose {
 	SpontaneousPayment(PaymentPreimage),
 }
 
+#[derive(Clone, Debug, PartialEq)]
+/// The reason which the channel was closed. See individual variants more details.
+pub enum ClosureReason {
+	/// Closure generated from receiving a peer error message.
+	///
+	/// Our counterparty may have broadcasted their latest commitment state, and we have
+	/// as well
+	CounterpartyForceClosed {
+		/// The error which the peer sent us.
+		///
+		/// The string should be sanitized before it is used (e.g emitted to logs
+		/// or printed to stdout). Otherwise, a well crafted error message may exploit
+		/// a security vulnerability in the terminal emulator or the logging subsystem.
+		peer_msg: String,
+	},
+	/// Closure generated from [`ChannelManager::force_close_channel`], called by the user.
+	HolderForceClosed,
+	/// The channel was closed after negotiating a cooperative close and we've now broadcasted
+	/// the cooperative close transaction. Note the shutdown may have been initiated by us.
+	//TODO: split between CounterpartyInitiated/LocallyInitiated
+	CooperativeClosure,
+	/// A commitment transaction was confirmed on chain, closing the channel. Most likely this
+	/// commitment transaction came from our counterparty, but it may also have come from
+	/// a copy of our own `ChannelMonitor`.
+	CommitmentTxBroadcasted,
+	/// Closure generated from processing an event, likely a HTLC forward/relay/reception.
+	ProcessingError {
+		err: String,
+	},
+	/// The `PeerManager` informed us that we've disconnected from the peer and that it is
+	/// unlikely we'll be able to connect to the peer.
```

Comment:
s/connect to the peer/connect to the peer again in the future. This may be caused by incompatible features which our counterparty, or we, require./